## 1-4 задачи решать с помощью функций, 5ю вручную
## Выберете тест и проверьте, есть ли различия между выборками:
## Задача 1

### 1 ) Даны две независимые выборки. Не соблюдается условие нормальности
#### x1 380,420, 290
#### y1 140,360,200,900
#### Сделайте вывод по результатам, полученным с помощью функции

Так как не соблюдается условие нормальности, выборки очень маленького объема, а также количество выборок равно 2 и они независимы, выбираем критерий Манна-Уитни.

In [16]:
import numpy as np
import scipy.stats as stats
X1= np.array([380,420, 290])
Y1= np.array ([140,360,200,900])
stats.mannwhitneyu(X1, Y1)




MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Принимаем alpha = 0.05. pvalue > alpha, поэтому у нас нет оснований отвергнуть нулевую гипотезу. Статистически значимых различий между выборками нет.

Расчет критерия Манна-Уитни:
380 140,360,200,900  4 -1 = 3
420 140,360,200,900  4 - 1 = 3
290 140,360,200,900  3 - 1 = 2
Критерий Манн-Уитни 3 + 3 + 2 = 8 
Расчеты совпали

## 2 ) Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?
1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150, 130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

Так как количество выборок равно 3 и они зависимы(люди одни и те же), выбираем критерий Фридмана.

In [18]:
A= np.array([150, 160, 165, 145, 155])

B= np.array([140, 155, 150, 130, 135])
 
C=np.array([130, 130, 120, 130, 125])

stats.friedmanchisquare(A, B, C)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Принимаем alpha = 0.05. pvalue < alpha, поэтому у нас есть основания для принятия альтернативной гипотезы. Мы получили статистически значимые различия между выборками. Препарат очевидно снижает давление.

## 3 ) Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.


In [ ]:
Так как количество выборок теперь равно 2 и они зависимы(люди одни и те же), выбираем критерий Уилкоксона.

In [31]:
stats.wilcoxon(A, B)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

Принимаем alpha = 0.05. pvalue > alpha, поэтому у нас нет оснований отвергнуть нулевую гипотезу. Мы не получили статистически значимых различий.

In [32]:
B - A

array([-10,  -5, -15, -15, -20])

Все числа отрицательные поэтому Критерий Уилкоксона равен 0. Что совпадает с решением функцией stats.wilcoxon. 
 

## 4) Даны 3 группы учеников плавания.
### В 1 группе время на дистанцию 50 м составляют:56, 60, 62, 55, 71, 67, 59, 58, 64, 67
### Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
### Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

Так как количество выборок равно 3 и они независимы(люди в каждой группе разные), выбираем критерий Крускала-Уоллиса.

In [33]:
gr1 =([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr2 = ([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr3 = ([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(gr1, gr2, gr3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

Принимаем alpha = 0.05. pvalue > alpha, поэтому у нас нет оснований отвергнуть нулевую гипотезу. Мы не получили статистически значимых различий. Статистически значимых различий по времени на дистанцию 50 м. между тремя группами необнаружены.

## 5) Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

Так как размеры изделий подчинены нормальному закону распределения, а дисперсия генеральной совокупности неизвестна используем критерий Стьюдента.


In [42]:
Size= np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
def find_average(S):
    Sum = 0
    for i in range(len(S)):
        Sum += S[i]
    return Sum/len(S)
X = find_average(Size)
X

2.5279999999999996

In [45]:
def find_std(S, X):
    Sum = 0
    for i in range(len(S)):
        Sum += (S[i] - X)**2
    return np.sqrt(Sum/(len(S) -1))
S = find_std(Size, X)
S

0.1572542173961923

Проверка:

In [50]:
S1 = np.std(Size, ddof = 1)
S1


0.1572542173961923

In [51]:
X1 = Size.mean()
X1

2.5279999999999996

Верно, значения найдены корректно.

In [56]:
T  = (X - 2.5)/(S/np.sqrt(10)) 
T

0.5630613661802959

По таблице критических точек распределения Стьюдента найдем критическую точку по уровню значимости α = 0,05 и числу степеней свободы k = 9 , откуда t ≈ 1.833. (При двустороннем тесте alpha = 0,025  t ≈ 2.262)
Так как T набл = 0.563<1.833 = t,   то можно принять нулевую гипотезу о том ,что изделия соответствуют заявленному размеру. 
ОТВЕТ: Статистически значимых различий нет.
